src/pyseekdb/client/client_base.py

```
@@ -1430,7 +2360,12 @@ def _collection_table_exists(self, table_name: str) -> bool:
 
     def _has_collection_v2(self, name: str) -> bool:
         try:
-            query_sql = f"SELECT COLLECTION_ID FROM {CollectionNames.sdk_collections_table_name()} WHERE COLLECTION_NAME = '{name}'"
+            name_escaped = escape_string(name)
+            query_sql = (
+                f"SELECT collection_id FROM {self._qtable(CollectionNames.sdk_collections_table_name())} "
+                f"WHERE collection_name = '{name_escaped}' "
+                f"ORDER BY created_at, collection_id LIMIT 1"
```

Ruff (0.15.18)

[error] Possible SQL injection vector through string-based query construction

(S608)